### ISSUE TITLE
Start-Process: Unable to load DLL 'api-ms-win-core-job-12-1-0.dll'

### ISSUE THREAD

**Issue by gradinkov** [PowerShell/PowerShell#6561]
Repro:

Use clean install of Windows 7 SP1 (apparently both x86 and x64 are affected). Then run the following command in PS 6.0.2 ZIP version:

~~~
Start-Process "ipconfig" -ArgumentList "/flushdns" -Wait -NoNewWindow
~~~

Result:

~~~
Start-Process : Unable to load DLL 'api-ms-win-core-job-l2-1-0.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
At line:1 char:1
+ Start-Process -FilePath "C:\7zip\7za.exe" -ArgumentList ("x -oC:\temp ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Start-Process], DllNotFoundException
+ FullyQualifiedErrorId : System.DllNotFoundException,Microsoft.PowerShell.Commands.StartProcessCommand
~~~

PowerShell 6.0.2 x86 zip
Windows 7 b7601 SP1 x86 (modern.ie VM)

Works perfectly fine on Win10 x64.

Also tried 6.1.0-preview1, behaves the same way.

**Comment by gradinkov** [comment]
Actually, way easier to repro:

~~~
Start-Process "ipconfig" -ArgumentList "/flushdns" -Wait -NoNewWindow
~~~

**Comment by iSazonov** [comment]
@gradinkov Please replace first repo with simple second in the issue description.

**Comment by iSazonov** [comment]
@mirichmo Could you please comment the Issue?

**Comment by gradinkov** [comment]
Sure thing, done!

**Comment by gradinkov** [comment]
The funny thing is that the commands actually run afterwards, just throw this exception first.

This is a problem when you run multiple commands in try-catch (as you normally should), which interrupts the rest of the try blocks and thus messes up execution.

TLDR if you run `start-process` in try-catch, anything that comes after start-process will be skipped.

**Comment by mirichmo** [comment]
@adityapatwardhan It looks like PowerShell Core needs another API set for Win 7. Do you know if there a way to scan for compliance (static analysis) rather than checking at runtime?

**Comment by adityapatwardhan** [comment]
The binary does not seem to be in the API set we use : `runtime.win7-x64.Microsoft.NETCore.Windows.ApiSets/1.0.1`. This needs further investigation.

**Comment by Arnavion** [comment]
I hit this on 6.0.1 on win7 x64.

```
$ $Error[0].exception.stacktrace

   at Microsoft.PowerShell.Commands.NativeMethods.CreateJobObject(IntPtr lpJobAttributes, String lpName)
   at Microsoft.PowerShell.Commands.ProcessCollection..ctor()
   at Microsoft.PowerShell.Commands.StartProcessCommand.BeginProcessing()
   at System.Management.Automation.Cmdlet.DoBeginProcessing()
   at System.Management.Automation.CommandProcessorBase.DoBegin()
```

**Comment by Arnavion** [comment]
@iSazonov Same thing with current master `GitCommitId: v6.1.0-preview.2-26-g84344cbb32ec5e2479e00394951832794bf37244`

```
Unable to load DLL 'api-ms-win-core-job-l2-1-0.dll' or one of its dependencies: The specified module could not be found. (Exception from HRESULT: 0x8007007E)
   at Microsoft.PowerShell.Commands.NativeMethods.CreateJobObject(IntPtr lpJobAttributes, String lpName)
   at Microsoft.PowerShell.Commands.ProcessCollection..ctor() in C:\Stuff\Sources\PowerShell\src\Microsoft.PowerShell.Commands.Management\commands\management\Process.cs:line 2551
   at Microsoft.PowerShell.Commands.StartProcessCommand.BeginProcessing() in C:\Stuff\Sources\PowerShell\src\Microsoft.PowerShell.Commands.Management\commands\management\Process.cs:line 2018
   at System.Management.Automation.Cmdlet.DoBeginProcessing() in C:\Stuff\Sources\PowerShell\src\System.Management.Automation\engine\cmdlet.cs:line 159
   at System.Management.Automation.CommandProcessorBase.DoBegin() in C:\Stuff\Sources\PowerShell\src\System.Management.Automation\engine\CommandProcessorBase.cs:line 499
```

**Comment by iSazonov** [comment]
@Arnavion Thanks for confirmation!

**Comment by meekaah** [comment]
any news? this is completely blocking us 👎 i still havent found a workaround
we're running windows server 2008 r2

**Comment by gradinkov** [comment]
The fact that after almost 4 months this issue is still open indeed is puzzling. Win7 is officially supported, yet, PowerShell Core is completely broken on it. That means around HALF of your userbase. I'd recommend to either declare Win7 unsupported or fix this critical blocker.

But I can't just say to my users that "yeah, your OS is theoretically supported, except in practice it won't work". I can't say "a fix is in the works" either coz I haven't seen any _real_ activity on the matter.

**Comment by iSazonov** [comment]
Looking https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-createjobobjecta
I guess we could directly reference Kernel32.dll

**Comment by adityapatwardhan** [comment]
I found a workaround for the issue. 

The issue is with `-Wait`. You can use `-PassThru` and wait for the process completion. Something like this:

```PowerShell
$process = Start-Process "ipconfig" -PassThru -NoNewWindow 
Wait-UntilTrue -sb { $process.HasExited }
```

The function `Wait-UntilTrue` is defined here:
https://github.com/PowerShell/PowerShell/blob/master/test/tools/Modules/HelpersCommon/HelpersCommon.psm1#L3

**Comment by meekaah** [comment]
Wait-UntilTrue : The term 'Wait-UntilTrue' is not recognized as the name of a cmdlet, function, script file, or... :( how are you using it? are you adding something else?

**Comment by adityapatwardhan** [comment]
@meekaah The function is defined the link I posted above. 

https://github.com/PowerShell/PowerShell/blob/master/test/tools/Modules/HelpersCommon/HelpersCommon.psm1#L3-L24

**Comment by fdge** [comment]
This error is slowing me down from moving From PowerShell to Core
This error happens on Server 2008 R2 but not on Windows 10 1709

**Comment by ferenczy** [comment]
One and a half years ago and the issue is still present. Could you let us know what's the plan with this bug, please?

Windows 7 Ultimate 64-bit
PowerShell Core 6.2.2
